### PR TITLE
MedicationRequest Transformer From MedicationStatement

### PR DESCRIPTION
**Pull request by taylorrassmannlibertyits**
https://vajira.max.gov/browse/API-1211 

Added Transformer for MedicationRequest.

Added /R4MedicationRequestFromMedicationStatementTransformer, unit test, and updated the MedicationRequestSample.

### CODE REVIEW

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medicationrequest/R4MedicationRequestFromMedicationStatementTransformer.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import lombok.Builder;
+import lombok.NonNull;
+
+@Builder
+public class R4MedicationRequestFromMedicationStatementTransformer {
+  @NonNull final DatamartMedicationStatement datamart;
+
+  static List<Dosage> dosageInstructionConverter(
+      Optional<Instant> effectiveDate, DatamartMedicationStatement.Dosage dosage) {
+    if (allBlank(dosage)) {
```
Comment (jhulbert-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medicationrequest/R4MedicationRequestFromMedicationStatementTransformer.java:27]: Looking at the rest of this method, we should check if dosage is null, and then do an `allBlank()` on everything else.

File: data-query/src/test/java/gov/va/api/health/dataquery/service/controller/medicationrequest/R4MedicationRequestFromMedicationStatementTransformerTest.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+
+import gov.va.api.health.dataquery.service.controller.medicationstatement.DatamartMedicationStatement;
+import gov.va.api.health.dataquery.service.controller.medicationstatement.MedicationStatementSamples;
+import gov.va.api.health.uscorer4.api.resources.MedicationRequest;
+import java.util.Optional;
+import org.junit.Test;
+
+public class R4MedicationRequestFromMedicationStatementTransformerTest {
+
```
Comment (jhulbert-va) [inline data-query/src/test/java/gov/va/api/health/dataquery/service/controller/medicationrequest/R4MedicationRequestFromMedicationStatementTransformerTest.java:12]: What about `note(Optional<String> note)`

File: data-query/src/test/java/gov/va/api/health/dataquery/service/controller/medicationrequest/R4MedicationRequestFromMedicationStatementTransformerTest.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  @Test
+  public void nullDosageInstructionTest() {
+    assertThat(
+            R4MedicationRequestFromMedicationStatementTransformer.dosageInstructionConverter(
+                Optional.empty(), null))
+        .isNull();
+  }
+
+  @Test
+  public void statusTests() {
```
Comment (jhulbert-va) [inline data-query/src/test/java/gov/va/api/health/dataquery/service/controller/medicationrequest/R4MedicationRequestFromMedicationStatementTransformerTest.java:22]: What about `Status.completed`?

File: data-query/src/test/java/gov/va/api/health/dataquery/service/controller/medicationrequest/R4MedicationRequestFromMedicationStatementTransformerTest.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
+package gov.va.api.health.dataquery.service.controller.medicationrequest;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import gov.va.api.health.dataquery.service.controller.medicationstatement.DatamartMedicationStatement;
+import gov.va.api.health.dataquery.service.controller.medicationstatement.MedicationStatementSamples;
+import gov.va.api.health.uscorer4.api.resources.MedicationRequest;
+import java.util.Optional;
+import org.junit.Test;
```
Comment (jhulbert-va) [inline data-query/src/test/java/gov/va/api/health/dataquery/service/controller/medicationrequest/R4MedicationRequestFromMedicationStatementTransformerTest.java:9]: [Junit5](https://blogs.oracle.com/javamagazine/migrating-from-junit-4-to-junit-5-important-differences-and-benefits?source=:em:nw:mt::RC_WWMK190726P00001:NSL400044496&elq_mid=159020&sh=2509022613082419122118221523291030330826&cmid=WWMK190726P00001C0008)

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medicationrequest/R4MedicationRequestFromMedicationStatementTransformer.java (hunk 1 above)
Comment (taylorrassmannlibertyits) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medicationrequest/R4MedicationRequestFromMedicationStatementTransformer.java:27]: Updated in next push.

File: data-query/src/test/java/gov/va/api/health/dataquery/service/controller/medicationrequest/R4MedicationRequestFromMedicationStatementTransformerTest.java (hunk 4 above)
Comment (taylorrassmannlibertyits) [reply data-query/src/test/java/gov/va/api/health/dataquery/service/controller/medicationrequest/R4MedicationRequestFromMedicationStatementTransformerTest.java:9]: Updated in next commit.

File: data-query/src/test/java/gov/va/api/health/dataquery/service/controller/medicationrequest/R4MedicationRequestFromMedicationStatementTransformerTest.java (hunk 3 above)
Comment (taylorrassmannlibertyits) [reply data-query/src/test/java/gov/va/api/health/dataquery/service/controller/medicationrequest/R4MedicationRequestFromMedicationStatementTransformerTest.java:26]: Updated in next push.

**Pull request merged**